From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mooziii <63669478+mooziii@users.noreply.github.com>
Date: Sat, 9 Jul 2022 14:32:49 +0200
Subject: [PATCH] Add biome limiter


diff --git a/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java b/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
index a8caf76ac93eb5f480773d4dd7e5c3784db429ac..eb9cf9a06faa51acd9fbfaebff78d151bf925e10 100644
--- a/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
+++ b/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
@@ -1,11 +1,13 @@
 package de.hglabor.papiergabel.config;
 
 import com.google.common.io.Files;
+import net.minecraft.world.level.biome.Biomes;
 import org.bukkit.configuration.file.YamlConfiguration;
 import java.io.File;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Field;
 import java.nio.charset.StandardCharsets;
+import java.util.List;
 
 @SuppressWarnings({"Unused", "Deprecation"})
 @Header("""
@@ -18,6 +20,14 @@ public class PapiergabelConfig {
     public static Messages messages = new Messages();
     public static Knockback knockback = new Knockback();
     public static Fixes fixes = new Fixes();
+    public static BiomeLimiter biomelimiter = new BiomeLimiter();
+
+    @MustSerialize
+    public static class BiomeLimiter {
+        public boolean enable = false;
+        public List<String> disabledBiomes = List.of("ocean");
+        public String replacementBiome = "plains";
+    }
 
     @MustSerialize
     public static class Fixes {
diff --git a/src/main/java/net/minecraft/world/level/biome/Biomes.java b/src/main/java/net/minecraft/world/level/biome/Biomes.java
index 06a65d823906c8cb2a80b431fe95e3c3a492a0f3..b410f4a423f0493dd530277d93d57e644e4705b0 100644
--- a/src/main/java/net/minecraft/world/level/biome/Biomes.java
+++ b/src/main/java/net/minecraft/world/level/biome/Biomes.java
@@ -1,5 +1,6 @@
 package net.minecraft.world.level.biome;
 
+import de.hglabor.papiergabel.config.PapiergabelConfig;
 import net.minecraft.core.Registry;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
@@ -70,6 +71,13 @@ public abstract class Biomes {
     public static final ResourceKey<Biome> END_BARRENS = register("end_barrens");
 
     private static ResourceKey<Biome> register(String name) {
+        // papiergabel Start
+        if (PapiergabelConfig.biomelimiter.enable) {
+            if (PapiergabelConfig.biomelimiter.disabledBiomes.contains(name)) {
+                return ResourceKey.create(Registry.BIOME_REGISTRY, new ResourceLocation(PapiergabelConfig.biomelimiter.replacementBiome));
+            }
+        }
+        // papergabel End
         return ResourceKey.create(Registry.BIOME_REGISTRY, new ResourceLocation(name));
     }
 }
