From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mooziii <63669478+mooziii@users.noreply.github.com>
Date: Sat, 9 Jul 2022 15:03:40 +0200
Subject: [PATCH] Fix biome limiter


diff --git a/src/main/java/de/hglabor/papiergabel/PapiergabelMain.java b/src/main/java/de/hglabor/papiergabel/PapiergabelMain.java
index 1d717f416b414c863129c7ebe06916e2fa264087..5862950c3e98ba5532a554a00e57d45a11dd1081 100644
--- a/src/main/java/de/hglabor/papiergabel/PapiergabelMain.java
+++ b/src/main/java/de/hglabor/papiergabel/PapiergabelMain.java
@@ -8,11 +8,7 @@ import java.io.File;
 
 public class PapiergabelMain {
 
-    public static void initPapiergabel() throws Exception {
-    }
-
-    public static void postLoad(MinecraftServer server) {
-        server.server.getCommandMap().register("papiergabel", "papiergabel", new PapiergabelCommand());
+    public static void initPapiergabel() {
         try {
             PapiergabelConfig.set(new PapiergabelConfig(), new File("papiergabel.yml"));
         } catch (Throwable t) {
@@ -20,4 +16,8 @@ public class PapiergabelMain {
         }
     }
 
+    public static void postLoad(MinecraftServer server) {
+        server.server.getCommandMap().register("papiergabel", "papiergabel", new PapiergabelCommand());
+    }
+
 }
diff --git a/src/test/java/io/papermc/paper/world/structure/ConfiguredStructureTest.java b/src/test/java/io/papermc/paper/world/structure/ConfiguredStructureTest.java
index 61efebe1d363b34e2043ccc4c6e28bb714c2fa31..db579891a693246ce3d7e126e8d900b96798d045 100644
--- a/src/test/java/io/papermc/paper/world/structure/ConfiguredStructureTest.java
+++ b/src/test/java/io/papermc/paper/world/structure/ConfiguredStructureTest.java
@@ -1,7 +1,6 @@
 package io.papermc.paper.world.structure;
 
 import io.papermc.paper.registry.Reference;
-import net.minecraft.data.BuiltinRegistries;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.Bootstrap;
diff --git a/src/test/java/org/bukkit/BiomeTest.java b/src/test/java/org/bukkit/BiomeTest.java
index 2ccc1ac837f1537f2f201f81aca24f744695a777..4d50862f6f4fc9b9d6ba356b0409aa629903b661 100644
--- a/src/test/java/org/bukkit/BiomeTest.java
+++ b/src/test/java/org/bukkit/BiomeTest.java
@@ -1,6 +1,5 @@
 package org.bukkit;
 
-import net.minecraft.data.BuiltinRegistries;
 import org.bukkit.block.Biome;
 import org.bukkit.craftbukkit.block.CraftBlock;
 import org.bukkit.support.AbstractTestingBase;
diff --git a/src/test/java/org/bukkit/craftbukkit/generator/ChunkDataTest.java b/src/test/java/org/bukkit/craftbukkit/generator/ChunkDataTest.java
index 690e2db9f75964b40ff20596d6b16b68d85884dc..ce739d8f060ae0c0f3ea316c999297b14078f1e0 100644
--- a/src/test/java/org/bukkit/craftbukkit/generator/ChunkDataTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/generator/ChunkDataTest.java
@@ -3,7 +3,6 @@ package org.bukkit.craftbukkit.generator;
 import static org.junit.Assert.*;
 
 import net.minecraft.core.Registry;
-import net.minecraft.data.BuiltinRegistries;
 import net.minecraft.world.level.biome.Biome;
 import org.bukkit.Material;
 import org.bukkit.block.data.BlockData;
