From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mooziii <63669478+mooziii@users.noreply.github.com>
Date: Sat, 9 Jul 2022 15:54:33 +0200
Subject: [PATCH] Fix biomelimiter again


diff --git a/src/main/java/com/destroystokyo/paper/antixray/ChunkPacketBlockControllerAntiXray.java b/src/main/java/com/destroystokyo/paper/antixray/ChunkPacketBlockControllerAntiXray.java
index a5d43901595f864c8a5dd5d013aa42d7f294e489..6409c0c75c27ccede2c3a19d3c80e09f00a49224 100644
--- a/src/main/java/com/destroystokyo/paper/antixray/ChunkPacketBlockControllerAntiXray.java
+++ b/src/main/java/com/destroystokyo/paper/antixray/ChunkPacketBlockControllerAntiXray.java
@@ -13,7 +13,6 @@ import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.server.level.ServerPlayerGameMode;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.EntityBlock;
diff --git a/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java b/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
index eb9cf9a06faa51acd9fbfaebff78d151bf925e10..bc0d1730702718cc4aeaaa0c6b74ed9683b5c624 100644
--- a/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
+++ b/src/main/java/de/hglabor/papiergabel/config/PapiergabelConfig.java
@@ -1,13 +1,11 @@
 package de.hglabor.papiergabel.config;
 
 import com.google.common.io.Files;
-import net.minecraft.world.level.biome.Biomes;
 import org.bukkit.configuration.file.YamlConfiguration;
 import java.io.File;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Field;
 import java.nio.charset.StandardCharsets;
-import java.util.List;
 
 @SuppressWarnings({"Unused", "Deprecation"})
 @Header("""
@@ -24,9 +22,9 @@ public class PapiergabelConfig {
 
     @MustSerialize
     public static class BiomeLimiter {
-        public boolean enable = false;
-        public List<String> disabledBiomes = List.of("ocean");
-        public String replacementBiome = "plains";
+        public boolean enableOceans = true;
+        public boolean enableDeepDark = true;
+        public boolean enableRivers = true;
     }
 
     @MustSerialize
diff --git a/src/main/java/net/minecraft/data/worldgen/SurfaceRuleData.java b/src/main/java/net/minecraft/data/worldgen/SurfaceRuleData.java
index 2f9ba60c0196c3722fbf5a44fbbcf22a85d438db..16721cb99a645e5d0fae1ad133c1ce57e61f7c9d 100644
--- a/src/main/java/net/minecraft/data/worldgen/SurfaceRuleData.java
+++ b/src/main/java/net/minecraft/data/worldgen/SurfaceRuleData.java
@@ -1,7 +1,6 @@
 package net.minecraft.data.worldgen;
 
 import com.google.common.collect.ImmutableList;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.levelgen.Noises;
diff --git a/src/main/java/net/minecraft/data/worldgen/biome/Biomes.java b/src/main/java/net/minecraft/data/worldgen/biome/Biomes.java
index aa8d3566a6979f2d61ebee409f98e2b9af0a6933..3ff4bc0e953fdd3b535698c9717d4fae5ca86fca 100644
--- a/src/main/java/net/minecraft/data/worldgen/biome/Biomes.java
+++ b/src/main/java/net/minecraft/data/worldgen/biome/Biomes.java
@@ -1,5 +1,6 @@
 package net.minecraft.data.worldgen.biome;
 
+import de.hglabor.papiergabel.config.PapiergabelConfig;
 import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
 import net.minecraft.data.BuiltinRegistries;
@@ -42,24 +43,30 @@ public abstract class Biomes {
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.FROZEN_PEAKS, OverworldBiomes.frozenPeaks());
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.JAGGED_PEAKS, OverworldBiomes.jaggedPeaks());
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.STONY_PEAKS, OverworldBiomes.stonyPeaks());
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.RIVER, OverworldBiomes.river(false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, OverworldBiomes.river(true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.BEACH, OverworldBiomes.beach(false, false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, OverworldBiomes.beach(true, false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.STONY_SHORE, OverworldBiomes.beach(false, true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.WARM_OCEAN, OverworldBiomes.warmOcean());
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.LUKEWARM_OCEAN, OverworldBiomes.lukeWarmOcean(false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_LUKEWARM_OCEAN, OverworldBiomes.lukeWarmOcean(true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.OCEAN, OverworldBiomes.ocean(false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_OCEAN, OverworldBiomes.ocean(true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.COLD_OCEAN, OverworldBiomes.coldOcean(false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_COLD_OCEAN, OverworldBiomes.coldOcean(true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, OverworldBiomes.frozenOcean(false));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, OverworldBiomes.frozenOcean(true));
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.MUSHROOM_FIELDS, OverworldBiomes.mushroomFields());
+        if (PapiergabelConfig.biomelimiter.enableRivers) {
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.RIVER, OverworldBiomes.river(false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, OverworldBiomes.river(true));
+        }
+        if (PapiergabelConfig.biomelimiter.enableOceans) {
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.BEACH, OverworldBiomes.beach(false, false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, OverworldBiomes.beach(true, false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.STONY_SHORE, OverworldBiomes.beach(false, true));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.WARM_OCEAN, OverworldBiomes.warmOcean());
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.LUKEWARM_OCEAN, OverworldBiomes.lukeWarmOcean(false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_LUKEWARM_OCEAN, OverworldBiomes.lukeWarmOcean(true));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.OCEAN, OverworldBiomes.ocean(false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_OCEAN, OverworldBiomes.ocean(true));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.COLD_OCEAN, OverworldBiomes.coldOcean(false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_COLD_OCEAN, OverworldBiomes.coldOcean(true));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, OverworldBiomes.frozenOcean(false));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, OverworldBiomes.frozenOcean(true));
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.MUSHROOM_FIELDS, OverworldBiomes.mushroomFields());
+        }
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DRIPSTONE_CAVES, OverworldBiomes.dripstoneCaves());
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.LUSH_CAVES, OverworldBiomes.lushCaves());
-        BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_DARK, OverworldBiomes.deepDark());
+        if (PapiergabelConfig.biomelimiter.enableDeepDark) {
+            BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.DEEP_DARK, OverworldBiomes.deepDark());
+        }
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.NETHER_WASTES, NetherBiomes.netherWastes());
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.WARPED_FOREST, NetherBiomes.warpedForest());
         BuiltinRegistries.register(registry, net.minecraft.world.level.biome.Biomes.CRIMSON_FOREST, NetherBiomes.crimsonForest());
diff --git a/src/main/java/net/minecraft/world/level/PathNavigationRegion.java b/src/main/java/net/minecraft/world/level/PathNavigationRegion.java
index 138f5576a1df6714e754d9829f209f1a1678413f..055df9df50e2d709540d01b71b7dc69e38c95aaf 100644
--- a/src/main/java/net/minecraft/world/level/PathNavigationRegion.java
+++ b/src/main/java/net/minecraft/world/level/PathNavigationRegion.java
@@ -9,10 +9,8 @@ import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
 import net.minecraft.core.SectionPos;
 import net.minecraft.util.profiling.ProfilerFiller;
-import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.biome.Biome;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
diff --git a/src/main/java/net/minecraft/world/level/biome/Biomes.java b/src/main/java/net/minecraft/world/level/biome/Biomes.java
deleted file mode 100644
index b410f4a423f0493dd530277d93d57e644e4705b0..0000000000000000000000000000000000000000
--- a/src/main/java/net/minecraft/world/level/biome/Biomes.java
+++ /dev/null
@@ -1,83 +0,0 @@
-package net.minecraft.world.level.biome;
-
-import de.hglabor.papiergabel.config.PapiergabelConfig;
-import net.minecraft.core.Registry;
-import net.minecraft.resources.ResourceKey;
-import net.minecraft.resources.ResourceLocation;
-
-public abstract class Biomes {
-    public static final ResourceKey<Biome> THE_VOID = register("the_void");
-    public static final ResourceKey<Biome> PLAINS = register("plains");
-    public static final ResourceKey<Biome> SUNFLOWER_PLAINS = register("sunflower_plains");
-    public static final ResourceKey<Biome> SNOWY_PLAINS = register("snowy_plains");
-    public static final ResourceKey<Biome> ICE_SPIKES = register("ice_spikes");
-    public static final ResourceKey<Biome> DESERT = register("desert");
-    public static final ResourceKey<Biome> SWAMP = register("swamp");
-    public static final ResourceKey<Biome> MANGROVE_SWAMP = register("mangrove_swamp");
-    public static final ResourceKey<Biome> FOREST = register("forest");
-    public static final ResourceKey<Biome> FLOWER_FOREST = register("flower_forest");
-    public static final ResourceKey<Biome> BIRCH_FOREST = register("birch_forest");
-    public static final ResourceKey<Biome> DARK_FOREST = register("dark_forest");
-    public static final ResourceKey<Biome> OLD_GROWTH_BIRCH_FOREST = register("old_growth_birch_forest");
-    public static final ResourceKey<Biome> OLD_GROWTH_PINE_TAIGA = register("old_growth_pine_taiga");
-    public static final ResourceKey<Biome> OLD_GROWTH_SPRUCE_TAIGA = register("old_growth_spruce_taiga");
-    public static final ResourceKey<Biome> TAIGA = register("taiga");
-    public static final ResourceKey<Biome> SNOWY_TAIGA = register("snowy_taiga");
-    public static final ResourceKey<Biome> SAVANNA = register("savanna");
-    public static final ResourceKey<Biome> SAVANNA_PLATEAU = register("savanna_plateau");
-    public static final ResourceKey<Biome> WINDSWEPT_HILLS = register("windswept_hills");
-    public static final ResourceKey<Biome> WINDSWEPT_GRAVELLY_HILLS = register("windswept_gravelly_hills");
-    public static final ResourceKey<Biome> WINDSWEPT_FOREST = register("windswept_forest");
-    public static final ResourceKey<Biome> WINDSWEPT_SAVANNA = register("windswept_savanna");
-    public static final ResourceKey<Biome> JUNGLE = register("jungle");
-    public static final ResourceKey<Biome> SPARSE_JUNGLE = register("sparse_jungle");
-    public static final ResourceKey<Biome> BAMBOO_JUNGLE = register("bamboo_jungle");
-    public static final ResourceKey<Biome> BADLANDS = register("badlands");
-    public static final ResourceKey<Biome> ERODED_BADLANDS = register("eroded_badlands");
-    public static final ResourceKey<Biome> WOODED_BADLANDS = register("wooded_badlands");
-    public static final ResourceKey<Biome> MEADOW = register("meadow");
-    public static final ResourceKey<Biome> GROVE = register("grove");
-    public static final ResourceKey<Biome> SNOWY_SLOPES = register("snowy_slopes");
-    public static final ResourceKey<Biome> FROZEN_PEAKS = register("frozen_peaks");
-    public static final ResourceKey<Biome> JAGGED_PEAKS = register("jagged_peaks");
-    public static final ResourceKey<Biome> STONY_PEAKS = register("stony_peaks");
-    public static final ResourceKey<Biome> RIVER = register("river");
-    public static final ResourceKey<Biome> FROZEN_RIVER = register("frozen_river");
-    public static final ResourceKey<Biome> BEACH = register("beach");
-    public static final ResourceKey<Biome> SNOWY_BEACH = register("snowy_beach");
-    public static final ResourceKey<Biome> STONY_SHORE = register("stony_shore");
-    public static final ResourceKey<Biome> WARM_OCEAN = register("warm_ocean");
-    public static final ResourceKey<Biome> LUKEWARM_OCEAN = register("lukewarm_ocean");
-    public static final ResourceKey<Biome> DEEP_LUKEWARM_OCEAN = register("deep_lukewarm_ocean");
-    public static final ResourceKey<Biome> OCEAN = register("ocean");
-    public static final ResourceKey<Biome> DEEP_OCEAN = register("deep_ocean");
-    public static final ResourceKey<Biome> COLD_OCEAN = register("cold_ocean");
-    public static final ResourceKey<Biome> DEEP_COLD_OCEAN = register("deep_cold_ocean");
-    public static final ResourceKey<Biome> FROZEN_OCEAN = register("frozen_ocean");
-    public static final ResourceKey<Biome> DEEP_FROZEN_OCEAN = register("deep_frozen_ocean");
-    public static final ResourceKey<Biome> MUSHROOM_FIELDS = register("mushroom_fields");
-    public static final ResourceKey<Biome> DRIPSTONE_CAVES = register("dripstone_caves");
-    public static final ResourceKey<Biome> LUSH_CAVES = register("lush_caves");
-    public static final ResourceKey<Biome> DEEP_DARK = register("deep_dark");
-    public static final ResourceKey<Biome> NETHER_WASTES = register("nether_wastes");
-    public static final ResourceKey<Biome> WARPED_FOREST = register("warped_forest");
-    public static final ResourceKey<Biome> CRIMSON_FOREST = register("crimson_forest");
-    public static final ResourceKey<Biome> SOUL_SAND_VALLEY = register("soul_sand_valley");
-    public static final ResourceKey<Biome> BASALT_DELTAS = register("basalt_deltas");
-    public static final ResourceKey<Biome> THE_END = register("the_end");
-    public static final ResourceKey<Biome> END_HIGHLANDS = register("end_highlands");
-    public static final ResourceKey<Biome> END_MIDLANDS = register("end_midlands");
-    public static final ResourceKey<Biome> SMALL_END_ISLANDS = register("small_end_islands");
-    public static final ResourceKey<Biome> END_BARRENS = register("end_barrens");
-
-    private static ResourceKey<Biome> register(String name) {
-        // papiergabel Start
-        if (PapiergabelConfig.biomelimiter.enable) {
-            if (PapiergabelConfig.biomelimiter.disabledBiomes.contains(name)) {
-                return ResourceKey.create(Registry.BIOME_REGISTRY, new ResourceLocation(PapiergabelConfig.biomelimiter.replacementBiome));
-            }
-        }
-        // papergabel End
-        return ResourceKey.create(Registry.BIOME_REGISTRY, new ResourceLocation(name));
-    }
-}
diff --git a/src/main/java/net/minecraft/world/level/chunk/LevelChunkSection.java b/src/main/java/net/minecraft/world/level/chunk/LevelChunkSection.java
index 5afb598d288d32877834cfb7d9796b334767286d..7348cfc59324c720f4b6c6b6b0e699d5f7382777 100644
--- a/src/main/java/net/minecraft/world/level/chunk/LevelChunkSection.java
+++ b/src/main/java/net/minecraft/world/level/chunk/LevelChunkSection.java
@@ -7,7 +7,6 @@ import net.minecraft.core.Registry;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.world.level.biome.Biome;
 import net.minecraft.world.level.biome.BiomeResolver;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.biome.Climate;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/ChunkSerializer.java b/src/main/java/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
index 3367c75b1c132b42465d4c355a6b5fd00c3efe23..1bd6b3ac39e3b9bc985825d880f3274c8bada11d 100644
--- a/src/main/java/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
+++ b/src/main/java/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
@@ -38,7 +38,6 @@ import net.minecraft.world.entity.ai.village.poi.PoiManager;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.LightLayer;
 import net.minecraft.world.level.biome.Biome;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.entity.BlockEntity;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
index 947b0b10fb965f30513fd0df5bc0910fb9cb9a71..1d304985781b110b4d971ed66e6b1c37bdb49c80 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
@@ -2,28 +2,21 @@ package org.bukkit.craftbukkit;
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Predicates;
-import com.mojang.serialization.Codec;
+
 import java.lang.ref.WeakReference;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
-import java.util.Objects;
-import java.util.concurrent.locks.LockSupport;
-import java.util.function.BooleanSupplier;
 import java.util.function.Predicate;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Holder;
 import net.minecraft.core.Registry;
 import net.minecraft.core.SectionPos;
-import net.minecraft.nbt.CompoundTag;
-import net.minecraft.nbt.NbtOps;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.util.thread.ProcessorMailbox;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.LightLayer;
 import net.minecraft.world.level.biome.Biome;
-import net.minecraft.world.level.biome.Biomes;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.chunk.ChunkAccess;
 import net.minecraft.world.level.chunk.ChunkStatus;
@@ -31,9 +24,6 @@ import net.minecraft.world.level.chunk.DataLayer;
 import net.minecraft.world.level.chunk.LevelChunkSection;
 import net.minecraft.world.level.chunk.PalettedContainer;
 import net.minecraft.world.level.chunk.PalettedContainerRO;
-import net.minecraft.world.level.chunk.storage.ChunkSerializer;
-import net.minecraft.world.level.chunk.storage.EntityStorage;
-import net.minecraft.world.level.entity.PersistentEntitySectionManager;
 import net.minecraft.world.level.levelgen.Heightmap;
 import net.minecraft.world.level.levelgen.WorldgenRandom;
 import net.minecraft.world.level.lighting.LevelLightEngine;
